### PR TITLE
Add KeyVault Token Capability for AccessToken auth

### PR DESCRIPTION
**Pull request by markcowl**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props')" />
```
Comment (maddieclayton) [inline src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj]: Do you need both of these xunit runner packages?

File: src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj [hunk 2]
```diff
@@ -16,7 +17,8 @@
     <TargetFrameworkProfile />
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
-    <NuGetPackageImportStamp>9fdcb6f4</NuGetPackageImportStamp>
```
Comment (maddieclayton) [inline src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj]: Is there a reason you are deleting this? If so, you likely should just remove the surrounding <NugetPackageImportStamp> tags.

File: src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj [hunk 3]
```diff
@@ -190,4 +192,10 @@
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets" Condition="Exists('..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
```
Comment (maddieclayton) [inline src/Common/Commands.Common.Authentication.Test/Commands.Common.Authentication.Test.csproj]: Also not sure if this is necessary.

File: src/Common/Commands.ScenarioTests.Common/Commands.ScenarioTests.Common.csproj [hunk 4]
```diff
@@ -15,7 +15,8 @@
     <FileAlignment>512</FileAlignment>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
-    <NuGetPackageImportStamp>64a73b29</NuGetPackageImportStamp>
```
Comment (maddieclayton) [inline src/Common/Commands.ScenarioTests.Common/Commands.ScenarioTests.Common.csproj]: Same comments on this file as the first.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj [hunk 5]
```diff
@@ -15,7 +15,8 @@
     <FileAlignment>512</FileAlignment>
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
-    <NuGetPackageImportStamp>e5bda5ba</NuGetPackageImportStamp>
```
Comment (maddieclayton) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj]: Same

**Comment by markcowl** [comment]
signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/70/

on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/302/

**Comment by maddieclayton** [comment]
@azuresdkci Test this please

File: src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs [hunk 6]
```diff
@@ -74,21 +74,23 @@ public string GetToken()
...
-                context.Account.Type != AzureAccount.AccountType.ServicePrincipal)
-                throw new ArgumentException(string.Format(KeyVaultProperties.Resources.UnsupportedAccountType, context.Account.Type));
-
-            if (context.Subscription != null && context.Account != null)
+            var tenantId = string.Empty;
+            if (context.Tenant != null && context.Tenant.GetId() != Guid.Empty)
+            {
+                tenantId = context.Tenant.Id.ToString();
+            }
+            else if (string.IsNullOrWhiteSpace(tenantId) && context.Subscription != null && context.Account != null)
```
Comment (maddieclayton) [inline src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs]: I don't think you need to check if the tenantId is empty - you just set it to empty above

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/303/
signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/73/
holding on merge until we have an end-to-end test

**Comment by markcowl** [comment]
@azuresdkci test this please

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Pull request merged**